### PR TITLE
test: browser test for invite decider flow

### LINKED ISSUE #2449: Browser test for invite-reviewer/-decider
split out from #2040

### PR DESCRIPTION
**Pull request by opqdonut**
on top of #2452

fixes #2449

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue

## Testing
- [x] valuable features are integration / browser / acceptance tested automatically

### CODE REVIEW

File: test/clj/rems/test_browser.clj [hunk 1]
```diff
@@ -549,6 +549,78 @@
...
+                                                                    "test-invite-decider"))
+    (test-helpers/submit-application (btu/context-get :application-id) "alice"))
+  (btu/with-postmortem
+    (testing "handler invites reviewer"
+      (login-as "developer")
+      (go-to-application (btu/context-get :application-id))
+      (btu/wait-visible {:tag :h1 :fn/has-text "test-invite-decider"})
+
+      (btu/scroll-and-click :request-decision-dropdown)
+      (Thread/sleep 100) ;; need to wait for dropdown to expand
```
Comment (Macroz) [inline test/clj/rems/test_browser.clj:569]: Why is this wait needed if you have `wait-visible` next?

File: test/clj/rems/test_browser.clj [hunk 2]
```diff
@@ -549,6 +549,78 @@
...
+      (btu/scroll-and-click :invite-decider)
+      (btu/wait-visible {:css ".alert-success"})
+      (logout))
+    (testing "get invite token"
+      (let [[token invitation] (-> (btu/context-get :application-id)
+                                   applications/get-application-internal
+                                   :application/invitation-tokens
+                                   first)]
+        (is (string? token))
+        (is (= {:application/decider {:name "Malice Applicant" :email "malice@example.com"}
```
Comment (Macroz) [inline test/clj/rems/test_browser.clj:585]: Why do you want to use malice and carl here? Is there some significance in using completely different values? Also could create a new user too.

File: test/clj/rems/test_browser.clj [hunk 3]
```diff
@@ -549,6 +549,78 @@
...
+                :event/actor "developer"}
+               invitation))
+        (btu/context-assoc! :token token)))
+    (testing "accept invitation"
+      (btu/go (str (btu/get-server-url) "application/accept-invitation/" (btu/context-get :token)))
+      (btu/wait-page-loaded)
+      (btu/wait-visible {:css ".login-btn"})
```
Comment (Macroz) [inline test/clj/rems/test_browser.clj:592]: We have this pattern but do we actually need the first `wait-page-loaded`?

**Review by Macroz — APPROVED** [review]

File: test/clj/rems/test_browser.clj (hunk 1 above)
Comment (opqdonut) [reply test/clj/rems/test_browser.clj:569]: It looked like the wait-visible on its own wasn't enough, but I guess the wait-visible is redundant now. I'll double check what's needed.

File: test/clj/rems/test_browser.clj (hunk 2 above)
Comment (opqdonut) [reply test/clj/rems/test_browser.clj:585]: Creating a fresh user makes sense, I'll do that

File: test/clj/rems/test_browser.clj (hunk 3 above)
Comment (opqdonut) [reply test/clj/rems/test_browser.clj:592]: maybe not, I'll check

Comment (opqdonut) [reply test/clj/rems/test_browser.clj]: removed

File: test/clj/rems/test_browser.clj (hunk 1 above)
Comment (opqdonut) [reply test/clj/rems/test_browser.clj]: removed

**Review by Macroz — APPROVED** [review]

**Pull request merged**